scripts/globals/teleports.lua

```diff
@@ -271,6 +278,21 @@ tpz.teleport.toHomeNation = function(player)
     end
 end
 
+-----------------------------------
+-- TELEPORT TO ALLIED NATION
+-----------------------------------
+
+tpz.teleport.toAlliedNation = function(player)
+    local Allegiance = player:getCampaignAllegiance()
+    if Allegiance == tpz.alliedNation.SANDORIA then
+        player:setPos(-98, 1, -41, 224, 80)
+    elseif Allegiance == tpz.alliedNation.BASTOK then
+        player:setPos(-291, -10 , -107, 212, 87)
+    elseif Allegiance == tpz.alliedNation.WINDURST then
+        player:setPos(-31.442, -5 , 129.202, 128, 94)
```

Comment:
I know the rest of this file sets a bad example, but does anyone have thoughts on this using the tables Nireya is defining above for the teleport destinations, rather than defining the same positions twice?

Nireya: `unpack(<table>)` is a lua method which takes a table and breaks it apart into multiple arguments. The order of both needs to line up - so it's not used too often - but it could theoretically be used in this case.

Reply:
I re-ordered the tables to line up to allow for this method,  but after taking a look at examples and reading I was still pretty stumped at how `unpack` works exactly

Reply:
An easy way to think about `pack` and `unpack` is that they add brackets or remove them.

```lua
some_table = pack(1, 2, 3, 4) -- results in a table of {1, 2, 3, 4}
unpack(some_table) -- results in a return value of 1, 2, 3, 4 as separate arguments
```

Unpack is useful in the sense that lua lets you assign multiple values at once, because functions can return multiple values by default.

```lua
local pos = {123, 456, 789}
x, y, z = unpack(pos)
```
Would result in x being assigned to 123, y to 456, and z to 789

Since you'd be calling setPos directly, you can unpack it in the function call:
```lua
local pos = destinations[[ids.BASTOK_MARKETS_S]]
player:setPos(unpack(pos))
```
Should result in a function call to `setPos` equivalent to:
`player:setPos(-291, -10 , -107, 212, 87)`